Fix typos

sampaccoud opened the pull request:
While reading documentation for the first time, I noticed a few typos and potential improvements.

sampaccoud committed:
👷(circleCI) Actually check availability of Docker images

There was a command to display in logs the list of images
but no actual test that the images had been created as
expected.

sampaccoud commented on .circleci/config.yml:
```diff
@@ -47,12 +45,19 @@ jobs:
+            image = bin/build $service
             bin/publish $service
+            image_name=$(echo $image | cut -d':' -f1)
+            image_tag=$(echo $image | cut -d':' -f2)
```
@cchaudier @jmaupetit do you know how to write these two lines in one?

cchaudier replied:
Yes, supress the second line. image_tag is unused ;)

sampaccoud commented on .circleci/config.yml:
```diff
@@ -12,7 +12,7 @@ jobs:
       - run:
           name: Ensure we have a clean docker environment
           command: |
-            docker images "fundocker/openshift*"
```
This statement was always true so I tried to fix it.

jmaupetit replied:
Yes, it was written only for logging/debug purpose. Testing it is a good idea! I would prefer having both for logging & test purpose. WDYT?

sampaccoud replied:
Good idea yes!

sampaccoud commented on .circleci/config.yml:
```diff
@@ -24,7 +24,9 @@ jobs:
       - run:
           name: Check build images
           command: |
-            docker images "fundocker/openshift*"
```
This statement was always true so I tried to fix it.

jmaupetit commented on .circleci/config.yml:
```diff
@@ -12,7 +12,7 @@ jobs:
           name: Ensure we have a clean docker environment
           command: |
-            docker images "fundocker/openshift*"
+            [ -z $(docker images -q "fundocker/openshift*") ]
```
Instead of the `[]` syntax you can use the `test` command that is more readable in this context:

```bash
test -z "$(docker images -q 'fundocker/openshift*')"
```

jmaupetit commented on .circleci/config.yml:
```diff
@@ -24,7 +24,9 @@ jobs:
-            docker images "fundocker/openshift*"
+            for service in $(ls docker/images); do
+              [ ! -z $(docker images -q "fundocker/openshift-${service}") ]
+            done
```
same remarks here

sampaccoud replied:
done

jmaupetit reviewed (commented):
Thank you for fixing my typos :wink:

sampaccoud committed:
✏️(documentation): fix typos and minor improvement to doc

Mostly typos but also mentioning DjangoCMS as one of the
applications we operate, not only Open edX any more... and
more to come!

jmaupetit commented on .circleci/config.yml:
```diff
@@ -25,6 +26,9 @@ jobs:
           name: Check build images
           command: |
             docker images "fundocker/openshift*"
+            for service in $(ls project/docker/images); do
```
Are you sure it's `project/docker/images` and not `docker/images`?

sampaccoud replied:
yes I'm sure because it was breaking. I sshed into the container in CircleCI and saw that "checkout" makes the clone inside a "project" directory. The build is green so it looks ok and I hope it's always the case...

jmaupetit replied:
I think after the checkout it `cd project`. Take a look at other `for` loops.

sampaccoud replied:
Ok you're right. I was confused because when I sshed it was at the root...
I added some logging so we can follow and be sure that it tested all services.

jmaupetit commented on .circleci/config.yml:
```diff
@@ -25,6 +26,10 @@ jobs:
             docker images "fundocker/openshift*"
+            for service in $(ls docker/images); do
+              ! test -z $(docker images -q fundocker/openshift-${service})
+              echo $service
```
What about printing a more complete message like:

```bash
echo "Image for service ${service} exists"
```

jmaupetit reviewed (approved):
:+1:
